components/Tabs/src/MDCTabBar.h

```diff
@@ -141,9 +141,10 @@ IB_DESIGNABLE
 
 /**
  Called before the selected item changes by user action. This method is not called for programmatic
- changes to the tab bar's selected item.
+ changes to the tab bar's selected item. Return YES to allow the selection.
+ If you don't implement all items changes are allowed.
  */
-- (void)tabBar:(nonnull MDCTabBar *)tabBar willSelectItem:(nonnull UITabBarItem *)item;
+- (BOOL)tabBar:(nonnull MDCTabBar *)tabBar shouldSelectItem:(nonnull UITabBarItem *)item;
```

Comment:
Unfortunately this is an API change that would affect behavior so we have to follow the correct process to ensure we don't break clients.

What this means here:
* We have to add the new method rather than replacing the old one and implement a backward-compatible behavior so existing clients are unaffected.
* The new method needs an API review. I think we can do this inline here.
* The old method should be deprecated following the documented process. See https://github.com/material-components/material-components-ios/blob/develop/contributing/deprecation_policy.md